python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -122,6 +123,18 @@ def select(
     return columns, new_ir, partition_info
 
 
+def _leaf_column_names(expr: Expr) -> tuple[str, ...]:
```

Comment:
A unit test for this would be handy :) I somehow manage to always mangle recursive functions every time I touch them.